Lesson9 (Unordered containers)/ForFileDirectory.cpp

```diff
@@ -0,0 +1,72 @@
+#include <iostream>
+#include <fstream>
+#include <experimental/filesystem>
+#include <algorithm>
+#include <unordered_map>
+#include <functional>
+#include <string>
+#include <map>
+#include "ForFileDirectory.h"
+namespace fs = std::experimental::filesystem;
+
+
+
+bool Key_Equal_my::operator()(const std::string& fileName1, const std::string& fileName2) const
+{
+    std::ifstream file1(fileName1, std::ifstream::binary|std::ifstream::ate);
+    std::ifstream file2(fileName2, std::ifstream::binary|std::ifstream::ate);
+
+
+    if (file1.fail() || file2.fail())
+    {
+        return false;
+    }
+    file1.seekg(0, std::ifstream::beg);
+    file2.seekg(0, std::ifstream::beg);
+    std::istreambuf_iterator<char> eos;
+    std::istreambuf_iterator<char> it1 (file1.rdbuf());
+    std::istreambuf_iterator<char> it2 (file2.rdbuf());
+    return std::equal(it1, eos, it2);
+}
+
+std::size_t Hash_my::operator() (std::string const& key) const
+{
+    std::size_t randomHash = 145;
```

Comment:
**You didn't get the hash function application.
Please read https://honest.engineering/posts/hash-functions-survival-guide.
Not accepted. Rework it!**